app/assets/javascripts/3dbio_viewer/src/webapp/components/nmr/NMRDialog.tsx

```diff
@@ -0,0 +1,373 @@
+        },
+        [setPage, setPageSize]
+    );
+
+    const onClick = React.useCallback(() => {
+        if (isSaving) return;
+        const cancel = save();
+        setSaving(_saving => cancel);
+    }, [isSaving, save]);
+
+    const stopSaving = React.useCallback(() => {
+        if (saving) {
+            saving();
+            hideSaving();
+        }
+    }, [saving, hideSaving]);
+
+    return (
+        <div style={styles.toolbar}>
+            <div style={styles.exportButton}>
+                <Button
+                    variant="outlined"
+                    disabled={isSaving}
+                    color="inherit"
+                    startIcon={<GetAppIcon />}
+                    size="small"
+                    onClick={onClick}
+                    style={{ opacity: isSaving ? 0.5 : 1 }}
+                >
+                    {i18n.t("Export all fragments")}
+                </Button>
+                {isSaving && (
+                    <div style={styles.exportStopButton} onClick={stopSaving}>
+                        <StyledCircularProgress size={20} />
+                        <StopIcon color="inherit" style={styles.stop} />
+                    </div>
+                )}
+            </div>
+            <TablePagination
+                component="div"
+                count={pagination.count}
+                page={pagination.page}
+                onPageChange={handleChangePage}
+                rowsPerPage={pagination.pageSize}
+                onRowsPerPageChange={handleChangeRowsPerPage}
+            />
+        </div>
+    );
+});
+
+interface DialogContentProps {
+    target: NMRFragmentTarget;
+    pagination: NMRPagination;
+}
+
+const DialogContent: React.FC<DialogContentProps> = React.memo(({ target, pagination }) => {
+    const headers = ["Name", "SMILES", "InchiKey", "Formula", "PubChem_ID", "Target", "Result"];
+    const Headers = headers.map((h, idx) => (
+        <TableCell align="left" key={idx}>
+            {h}
+        </TableCell>
+    ));
+
+    const Items = _.sortBy(target.fragments, f => !f.binding).map((fragment, idx) => {
+        const {
+            binding,
+            ligand: { name: ligandName, smiles, inChI, formula, pubchemId },
+        } = fragment;
+
+        return (
+            <StyledTableRow key={idx} binding={binding}>
+                <TableCell>{pagination.pageSize * pagination.page + idx + 1}</TableCell>
+                <TableCell>{ligandName}</TableCell>
+                <TableCell align="left">{smiles}</TableCell>
+                <TableCell align="left">{inChI}</TableCell>
+                <TableCell align="left">{formula}</TableCell>
+                <TableCell align="left">{pubchemId}</TableCell>
+                <TableCell align="left">{target.name}</TableCell>
+                <TableCell align="left">
+                    {binding ? i18n.t("Binding") : i18n.t("Not binding")}
+                </TableCell>
+            </StyledTableRow>
+        );
+    });
```

Comment:
That's not very idiomatic. You'd create a separate React component and use it within the render return.